README.md

```diff
@@ -1,24 +1,161 @@
+
+```ruby
+# app/controllers/posts_controller.rb
+
+class PostsController < ApplicationController
+  def show
+  end
+end
+```
+
+Running the tests now you will get a failure saying that that we have a missing view template, let's fix that by creating a view file that is mapped to our ```show``` action by running the following terminal command (we also need to create a posts directory): ```mkdir app/views/posts && touch app/views/posts/show.html.erb```
+
+Running the tests now and we're all green, which means that the request will be properly routes through the controller and view and returns a HTTP status code of ```200```.
+
+Now that we have the routing configured, let's build a test to see if the post content is rendered on the show page with the title being in an ```h1``` tag and the description in a ```p``` tag, below are the scenarios (if you reference the initial spec setup, ```subject``` holds the post that we created with FactoryGirl):
+
+```ruby
+
+# Post factory for reference:
+# title "My Great Post"
+# description "My amazing description"
+
+scenario 'title is shown on the show page in a h1 tag' do
+  visit "/posts/#{subject.id}"
+  expect(page).to have_css("h1", text: "My Great Post")
+end
+
+scenario 'description is shown on the show page in a p tag' do
+  skip
+end
+```
+
+This gives us a failure that says: ```expected to find css "h1" with text "My Great Post" but there were no matches```. To fix this we first need to capture the post's id in the controller's show action, let's do that here:
+
+```ruby
+# app/controllers/posts_controller.rb
+
+def show
+  @post = Post.find(params[:id])
+end
+```
+
+In this line, our show action is running a database query on the Post model that will return a post that has the id that matches the route parameters, and it will store this record in the ```@post``` instance variable and make it available to the ```show.html.erb``` file. Let's get our spec passing by placing the post's title on the show view template:
+
+```ERB
+<% # app/views/posts/show.html.erb %>
+<h1><%= @post.title %></h1>
+```
+
+We're back to all green! Now let's implement the description spec:
+
+```ruby
+scenario 'description is shown on the show page in a p tag' do
+  visit "/posts/#{subject.id}"
+  expect(page).to have_css("p", text: "My amazing description")
+end
+```
+This will give us a failure since there are no matches on the template yet, to implement this fix, update the view:
+
+```ERB
+<% # app/views/posts/show.html.erb %>
+<h1><%= @post.title %></h1>
+<p><%= @post.description %></p>
+```
+
+Now we're all passing, you now know how to create dynamic routes in Rails! However we would be remiss if we didn't follow the full "Red, Green, Refactor" TDD workflow. There are a few elements of the application that can be refactored. First let's add some syntactic sugar to our route file.
```

Comment:
They don't really know the term "syntactic sugar" yet. So I'd just say, "instead of doing that long drawn out `get` route in our `routes.rb` file, we can use ruby's RESTful defaults and use the `resources` method. Problem is, we only have one of the seven RESTful routes. Thankfully, if we pass in another option, the `only` option we can choose which of the seven RESTful routes we care about. In this case we only care about the `show` action"